packages/react-devtools-extensions/src/contentScripts/proxy.js

```diff
@@ -2,23 +2,17 @@
 
 'use strict';
 
+window.addEventListener('unload', function ({target}) {
```

Comment:
Is this going to break the bf-cache for pages even when the devtools is not open? https://developer.mozilla.org/en-US/docs/Web/API/Window/unload_event#usage_notes

Reply:
Yes, but only this content script won't be put in `bfcache`. This is our goal, because if content scripts in Firefox are not unloaded / loaded properly again, extension might not receive some important events that are necessary to start

Reply:
Are you sure that it's just the script not ending up in bf-cache?

[MDN says](https://developer.mozilla.org/en-US/docs/Web/API/Window/unload_event#usage_notes):
> To combat this, some browsers (such as Firefox) will not place pages in the bfcache if they have unload listeners, and this is bad for performance.

Reply:
Yeah, I interpret
>will not place pages in the bfcache if they have unload listeners, and this is bad for performance.
as "if page has unload listeners, it won't be in bfcache", this doesn't mean that every page won't be in bfcache.

Content scripts are loaded separately from "page", and I think for Firefox it has a distinct page, so it won't be cached, but page, which runs react application, will. I will double-check this though.

Reply:
lgtm once you double check

Reply:
Looks like it really breakes bfcache for Firefox. I've moved to [pagehide](https://developer.mozilla.org/en-US/docs/Web/API/Window/pagehide_event) / pageshow events, they are [bfcache-compatible](https://developer.mozilla.org/en-US/docs/Web/API/Window/pagehide_event#usage_notes). Thanks again, re-requesting review.